server/src/main/java/org/opensearch/indices/ShardLimitValidator.java

```diff
@@ -66,6 +68,14 @@ public class ShardLimitValidator {
         Setting.Property.NodeScope
     );
 
+    public static final Setting<Integer> SETTING_CLUSTER_MAX_SHARDS_PER_CLUSTER = Setting.intSetting(
+        SETTING_MAX_SHARDS_PER_CLUSTER,
+        -1,
+        -1,
```

Comment:
Should we change the minValue to be 1. I understand the negative value, but would a value of 0 make sense here? May be you can add validation around possible values being -1 and all positive integers.

Reply:
-1 represent that the setting value is unset. This pattern is followed in [multiple existing settings](https://github.com/opensearch-project/OpenSearch/blob/53d54de90423e31af7ca2514b953d77df4ac2be4/server/src/main/java/org/opensearch/cluster/routing/allocation/decider/ShardsLimitAllocationDecider.java#L79), so did not modified the behaviour here as well.